### DISCUSSION TITLE
Make it possible to build a DLL without DllMain

### DISCUSSION THREAD

**Discussion by ghost** [LADSoft/OrangeC#1002] [Ideas]
This is possible with GCC compiler.

**Comment by GitMensch** [comment]
This is also possible with MSVC and OrangeC, just compile. Do you have an error which shows different?

**Reply by ghost** [reply]
His answer here confused me: https://github.com/LADSoft/OrangeC/discussions/981#discussioncomment-7559444

> for a dll do this:
> 
> occ -Wd testprog.c
> 
> and here you need DllMain

This implied that I need DllMain.

**Comment by LADSoft** [comment]
so I misspoke, what I probably should have said is if you need a main use DllMain().    I did check a minute ago, OrangeC is quite happy making a DLL if you don't explicitly add a DllMain().
